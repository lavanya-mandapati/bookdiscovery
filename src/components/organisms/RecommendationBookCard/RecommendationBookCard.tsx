import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import Divider from '@material-ui/core/Divider';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import BookmarkIcon from '@material-ui/icons/Bookmark';
import PlaylistAddCheckIcon from '@material-ui/icons/PlaylistAddCheck';
import Rating from '@material-ui/lab/Rating';
import ReplySharpIcon from '@material-ui/icons/ReplySharp';
import { CardActions, TableCell, Typography } from '@material-ui/core';
import { COLORS } from '../../../theme/constants';
import PlaylistAddIcon from '@material-ui/icons/PlaylistAddRounded';
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';
import theme from '../../../theme/theme';
import { useState } from 'react';

const useStyles = makeStyles({
    root: {
        width: "255px",
        height: "331px",
        padding:theme.spacing(5,0,2),
        marginLeft: theme.spacing(4),
        borderRadius: "4px",
        border: "solid 1px",
        borderColor: COLORS.GREY_300
    },
    image: {
        width: "180px",
        height: "190px",
        margin: theme.spacing(0, 6, 2, 7),
        objectFit: "contain"
    },
    title: {
        width: "216px",
        height: "28px",
        margin: theme.spacing(1, 8),
        color: COLORS.CHARCOAL_GREY
    },
    author: {
        width: "46px",
        height: "22px",
        textTransform: "none",
        color: COLORS.CHARCOAL_GREY_100
    },
    authorName: {
        height: "24px",
        float: "right"
    },
    star:{
        padding:theme.spacing(0)
    },
    rating: {
        width: "108px",
        height: "22px",
        margin: theme.spacing(1,2),
        textTransform: "none",
        textAlign: "left"
    },
    actions: {
        padding: theme.spacing(0),
        marginLeft: theme.spacing(8),
    },
    titleActions: {
        padding: theme.spacing(0)
    },
    bookmark: {
        color: COLORS.PRIMARY_COLOR_500,
        padding: theme.spacing(2),
        fontSize: "22px",
    },
    bookmarkBorder: {
        color: COLORS.ICON_500,
        padding: theme.spacing(2),
        fontSize: "22px",
    },
    reply: {
        color: COLORS.GREY_500,
        padding: theme.spacing(1, 2),
        fontSize: "22px"
    },
    playlist: {
        color: COLORS.GREY_500,
        padding: theme.spacing(1, 2),
        fontSize: "22px"
    },
    content:{
        padding: theme.spacing(0)
    },
    cell: {
        padding: theme.spacing(1),
        borderBottom: "none",
        width: "20px"
      }
});

export type RecommendationBookCardProps = {
    title: string;
    authorName: string;
    image: string;
}

const RecommendationBookCard: React.FC<RecommendationBookCardProps> = (props) => {
    const [bookMark, setBookMark] = useState(false);
    const bookmarkClick = () => {
        setBookMark((prev) => {
            return !prev;
        });
    }
    const [playlistAdded, setPlaylistAdded] = useState(false);
    const playlistClick = () => {
        setPlaylistAdded((prev) => {
            return !prev;
        });
    }
    const onClick = () => {
        return alert("Directed to BookDetail Page");
    }
    const classes = useStyles();
    return (
        <Card className={classes.root}>
            <TableCell className={classes.cell}>
                <CardMedia className={classes.image}
                    component="img"
                    image={props.image}
                    onClick={onClick} />
            </TableCell>
            <CardContent className={classes.content}>
                <CardActions className={classes.titleActions} >
                    <Typography className={classes.title} variant="h6">{props.title}</Typography>
                </CardActions>
                <CardActions className={classes.actions}>
                    <Typography className={classes.author} variant="body2">Author</Typography>
                    <Typography className={classes.authorName} variant="subtitle2">{props.authorName}</Typography>
                </CardActions>
                <CardActions className={classes.actions}>
                    <Rating size="small" read-only value={4} readOnly />
                    <Typography className={classes.star} variant="body1">4.0</Typography>
                    <Typography className={classes.rating} variant="overline">(1,204 ratings)</Typography>
                </CardActions>
            </CardContent>
            <Divider />
            <CardActions className={classes.actions}>
                {bookMark && <BookmarkIcon className={classes.bookmark} onClick={bookmarkClick} />}
                {!bookMark && <BookmarkBorderIcon className={classes.bookmarkBorder} onClick={bookmarkClick} />}
                <ReplySharpIcon className={classes.reply} />
                {!playlistAdded && <PlaylistAddIcon className={classes.playlist} onClick={playlistClick} />}
                {playlistAdded && <PlaylistAddCheckIcon className={classes.playlist} onClick={playlistClick} />}
            </CardActions>

        </Card>
    );
}

export default RecommendationBookCard;