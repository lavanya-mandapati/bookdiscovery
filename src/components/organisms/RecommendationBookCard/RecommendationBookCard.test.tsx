import { render, cleanup } from '@testing-library/react'
import RecommendationBookCard from './RecommendationBookCard'

afterEach(cleanup);

describe('RecommdationBookCard Testing', () => {

    it.only('Matches Snapshot', () => {
        const { container } = render(<RecommendationBookCard title="Nature Chemistry"
        authorName="J D Lee"
        image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png" />);
        expect(container).toMatchSnapshot();
    });


    it('Should render image correctly', () => {
        const { container } = render(<RecommendationBookCard title="Nature Chemistry"
        authorName="J D Lee"
        image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png" />);
        expect(container.children).toBeInTheDocument();
    });
})