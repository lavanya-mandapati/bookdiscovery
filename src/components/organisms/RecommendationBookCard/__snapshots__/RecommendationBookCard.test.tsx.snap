// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommdationBookCard Testing Matches Snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <td
      class="MuiTableCell-root makeStyles-cell-15"
    >
      <img
        class="MuiCardMedia-root makeStyles-image-2 MuiCardMedia-media MuiCardMedia-img"
        src="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png"
      />
    </td>
    <div
      class="MuiCardContent-root makeStyles-content-14"
    >
      <div
        class="MuiCardActions-root makeStyles-titleActions-9 MuiCardActions-spacing"
      >
        <h6
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          Nature Chemistry
        </h6>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-8 MuiCardActions-spacing"
      >
        <p
          class="MuiTypography-root makeStyles-author-4 MuiTypography-body2"
        >
          Author
        </p>
        <h6
          class="MuiTypography-root makeStyles-authorName-5 MuiTypography-subtitle2"
        >
          J D Lee
        </h6>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-8 MuiCardActions-spacing"
      >
        <span
          aria-label="4 Stars"
          class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
          role="img"
        >
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root makeStyles-star-6 MuiTypography-body1"
        >
          4.0
        </p>
        <span
          class="MuiTypography-root makeStyles-rating-7 MuiTypography-overline"
        >
          (1,204 ratings)
        </span>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiCardActions-root makeStyles-actions-8 MuiCardActions-spacing"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-bookmarkBorder-11"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-reply-12"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-playlist-13"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 10H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm5 8v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-3c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3zM3 16h6c.55 0 1-.45 1-1s-.45-1-1-1H3c-.55 0-1 .45-1 1s.45 1 1 1z"
        />
      </svg>
    </div>
  </div>
</div>
`;
